### DISCUSSION TITLE
how to specify highlight.js theme in css

### DISCUSSION THREAD

**Discussion by infinitesteps** [marp-team/marp#304] [Q&A] [answered]
When creating a CSS theme, how do I specify the highlight.js style and what styles are available? Based on the syntax in gaia.scss I tried the following:

```css
@import '~highlight.js/styles/zenburn';
```

But that didn't seem to work.

**Comment by yhatt** [comment, marked as answer]
>  `~highlight.js` is a syntax of [sass-loader](https://github.com/webpack-contrib/sass-loader#resolving-import-at-rules)/[rollup-plugin-postcss](https://github.com/egoist/rollup-plugin-postcss#imports) to resolve the path of `node_modules`. Marp Core is using [rollup-plugin-postcss to build and bundle themes](https://github.com/marp-team/marp-core/blob/3ae79e06953bb4baf26ded4c9e1c5154169d70a4/rollup.config.js#L30-L52).
>
> _&mdash; https://github.com/marp-team/marp/discussions/73#discussioncomment-1381499_

Setting up Sass preprocessor (and tools in around) is required to import highlight.js theme through `@import` correctly. The compiled theme CSS will replace `@import` to its contents. Marp ecosystem is always using the compiled themes.

Or you can pick a theme from highlight.js source and put raw styles to your theme directly.
https://github.com/highlightjs/highlight.js/tree/main/src/styles

---

FYI: We know highlight.js theming is one of confusable things in Marp theme, so have planned a new highlight system at marp-team/marp-core#296. You may have to follow up the update of code highlight in the future (but I promise changing highlight colors is going to become simpler than current :)

- Related
  - https://github.com/marp-team/marp/discussions/103
  - https://github.com/marp-team/marp-core/issues/296
